### PR TITLE
[FLINK-6781] Make statement fetch size configurable in JDBCInputFormat

### PR DESCRIPTION
**Pull request by mbode**

### CODE REVIEW

File: flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java [hunk 1]
```diff
@@ -101,6 +105,33 @@ public void testIncompleteConfiguration() throws IOException {
...
 	@Test
+	public void defaultFetchSizeIsUsedIfNotConfiguredOtherwise() throws SQLException {
+		jdbcInputFormat = JDBCInputFormat.buildJDBCInputFormat()
+			.setDrivername(DRIVER_CLASS)
+			.setDBUrl(DB_URL)
+			.setQuery(SELECT_ALL_BOOKS)
+			.setRowTypeInfo(ROW_TYPE_INFO)
+			.finish();
+		jdbcInputFormat.openInputFormat();
+		assertThat(jdbcInputFormat.getStatement().getFetchSize(), equalTo(1));
```
Comment (zentol) [inline flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java]: That is equals 1 is an internal detail of the derby jdbc driver is it not? Not sure how useful this test is.

File: flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java [hunk 2]
```diff
@@ -378,6 +388,11 @@ public JDBCInputFormatBuilder setRowTypeInfo(RowTypeInfo rowTypeInfo) {
 			return this;
 		}
 
+		public JDBCInputFormatBuilder setFetchSize(int fetchSize) {
+			format.fetchSize = fetchSize;
```
Comment (zentol) [inline flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormat.java]: let's add a `Preconditions` check that fetchSize is greater than 0.

File: flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java [hunk 3]
```diff
@@ -101,6 +105,33 @@ public void testIncompleteConfiguration() throws IOException {
...
+
+	}
+
+	@Test
+	public void fetchSizeCanBeConfigured() throws SQLException {
```
Comment (zentol) [inline flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java]: by convention the name of test methods should be prefixed with "test".

File: flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java [hunk 4]
```diff
@@ -101,6 +105,33 @@ public void testIncompleteConfiguration() throws IOException {
...
+		final int desiredFetchSize = 10_000;
+		jdbcInputFormat = JDBCInputFormat.buildJDBCInputFormat()
+			.setDrivername(DRIVER_CLASS)
+			.setDBUrl(DB_URL)
+			.setQuery(SELECT_ALL_BOOKS)
+			.setRowTypeInfo(ROW_TYPE_INFO)
+			.setFetchSize(desiredFetchSize)
+			.finish();
+		jdbcInputFormat.openInputFormat();
+		assertThat(jdbcInputFormat.getStatement().getFetchSize(), equalTo(desiredFetchSize));
```
Comment (zentol) [inline flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java]: Would be good to replace these with junit assertEquals, to be consistent with the remaining tests in flink.

File: flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java (hunk 1 above)
Comment (mbode) [inline flink-connectors/flink-jdbc/src/test/java/org/apache/flink/api/java/io/jdbc/JDBCInputFormatTest.java]: The idea was to make sure that existing behavior is not broken. I have made the dependency on the fact that derby is used in this test explicit.

**Comment by mbode** [comment]
[Green Travis](https://travis-ci.org/mbode/flink/builds/238207585)

**Comment by zentol** [comment]
Looks good, +1 to merge.

**Comment by zentol** [comment]
merging.

**Pull request closed without merging**